api/v1/lib/httpcli/http.go

```diff
@@ -54,21 +57,45 @@ var (
 	}
 
 	defaultErrorMapper = ErrorMapperFunc(func(code int) error {
+		// for now, just scrape the string of the deprecated error var and use that
+		// as the APIError.Message. Eventually we'll get rid of the Err* variables.
+		// TODO(jdef) simplify this after v0.0.3
 		err, ok := codeErrors[code]
 		if !ok {
-			err = ProtocolError(code)
+			err = &APIError{Code: code}
+		}
+		return &APIError{
+			Code:    code,
+			Message: err.Error(),
 		}
-		return err
 	})
 )
 
 // ProtocolError is a generic error type returned for expected status codes
 // received from Mesos.
+// Deprecated: no longer used in favor of APIError.
```

Comment:
Might be better to just break the API now and remove this, otherwise downstream consumers will wonder why they're not seeing ProtocolError's anymore. Better to break the build and force change.

Maybe tackle this in a follow-up PR? This one is getting a bit long already.